[Writing Tools] Send entire contents as context to Writing Tools for Compose composition type

rr-codes opened the pull request:
#### ceb2721f3072abc286af88e9d1f124e3cfc1fc03
<pre>
[Writing Tools] Send entire contents as context to Writing Tools for Compose composition type
<a href="https://bugs.webkit.org/show_bug.cgi?id=278766">https://bugs.webkit.org/show_bug.cgi?id=278766</a>
<a href="https://rdar.apple.com/131875217">rdar://131875217</a>

Reviewed by Abrar Rahman Protyasha.

For the Compose composition type sessions, send the entire range of the editable content instead
of just the enclosing paragraph(s).

* Source/WebCore/page/writing-tools/WritingToolsController.mm:
(WebCore::contextRangeForSession):
(WebCore::WritingToolsController::willBeginWritingToolsSession):
(WebCore::contextRangeForDocument): Deleted.
* Source/WebCore/page/writing-tools/WritingToolsTypes.h:
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/UIProcess/Cocoa/PlatformWritingToolsUtilities.mm:
(WebKit::convertToWebCompositionSessionType):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WritingTools.mm:
(TEST(WritingTools, ShowAffordance)):
(TEST(WritingTools, CompositionWithComposeCompositionType)):

Canonical link: <a href="https://commits.webkit.org/282859@main">https://commits.webkit.org/282859@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/830cf67c11d03a12ba8c8fc5e68afdf304beb8c3

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/64424 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/43788 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/17020 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/68445 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/15031 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/66543 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/51503 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/15311 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/51839 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/10368 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/67492 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/40481 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/55744 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/32458 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/37150 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/13122 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/13905 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/59111 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/13451 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/70146 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/8371 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/12963 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/59162 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/8405 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/55833 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/59329 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/6917 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/597 "Passed tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/9776 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/39602 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/40680 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/41863 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/40423 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas commented on Source/WebCore/page/writing-tools/WritingToolsController.mm:
```diff
@@ -104,7 +104,7 @@
-static std::optional<SimpleRange> contextRangeForDocument(const Document& document)
+static std::optional<SimpleRange> contextRangeForSession(const Document& document, const std::optional<WritingTools::Session>& session)
 {
     // If the selection is a range, the range of the context should be the range of the paragraph
```
Is this comment still accurate?

rr-codes replied:
it is for all cases except smart replies and compose, in which case I have separate comments for those

rr-codes committed:
[Writing Tools] Send entire contents as context to Writing Tools for Compose composition type
https://bugs.webkit.org/show_bug.cgi?id=278766
rdar://131875217

Reviewed by Abrar Rahman Protyasha.

For the Compose composition type sessions, send the entire range of the editable content instead
of just the enclosing paragraph(s).

* Source/WebCore/page/writing-tools/WritingToolsController.mm:
(WebCore::contextRangeForSession):
(WebCore::WritingToolsController::willBeginWritingToolsSession):
(WebCore::contextRangeForDocument): Deleted.
* Source/WebCore/page/writing-tools/WritingToolsTypes.h:
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/UIProcess/Cocoa/PlatformWritingToolsUtilities.mm:
(WebKit::convertToWebCompositionSessionType):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WritingTools.mm:
(TEST(WritingTools, ShowAffordance)):
(TEST(WritingTools, CompositionWithComposeCompositionType)):

Canonical link: https://commits.webkit.org/282859@main
